lib/praxis/mapper/resource.rb

```diff
@@ -79,6 +89,130 @@ def self.define_model_accessors
         end
       end
 
+      def self.hookup_callbacks
+        return unless ancestors.include?(Praxis::Mapper::Resources::Callbacks)
+
+        affected_methods = (before_callbacks.keys + after_callbacks.keys + around_callbacks.keys).uniq
+        # TODO!! Only create 1 prepended module for all methods!!!
+        affected_methods&.each do |method|
+          calls = {}
+          calls[:before] = before_callbacks[method] if before_callbacks.key?(method)
+          calls[:around] = around_callbacks[method] if around_callbacks.key?(method)
+          calls[:after] = after_callbacks[method] if after_callbacks.key?(method)
+
+          if method_defined?(method)
```

Comment:
Worth checking for the case where `method` is both class and instance, and raising vs. installing a hook on the instance? Probably a corner case, but, could save frustration.

Reply:
I've changed that so you can explicitly define class or instance methods in the callbacks. Thanks for bringing it up